controller/src/main/java/io/pravega/controller/server/rpc/grpc/impl/GRPCServerConfigImpl.java

```diff
@@ -66,13 +68,18 @@ public GRPCServerConfigImpl(final int port, final String publishedRPCHost, final
                     "accessTokenTtlInSeconds should be -1 (token never expires), 0 (token immediately expires) "
                             + "or a positive integer representing the number of seconds after which the token expires.");
         }
+        if (tlsEnabled && (tlsProtocolVersion != null)) {
```

Comment:
Having a _class_:
```
public class TLSProtocolVersion {
    private String[] tlsProtocols;

    public TLSProtocolVersion(String s) {
        tlsProtocols = TlsProtocolVersion.parse(s);
    }

    public String[] tlsProtocols() {
        return tlsProtocols;
    }

    public boolean isValid() {
        return tlsProtocols == null;
    }

    public enum TlsProtocolVersion {
        TLSv1_2("TLSv1.2"),
        TLSv1_3("TLSv1.3"),
        TLS1_2ANDTLS1_3("TLSv1.2,TLSv1.3"),
        TLS1_3ANDTLS1_2("TLSv1.3,TLSv1.2");

        String protocol;

        TlsProtocolVersion(String tlsprotocol) {
            protocol = tlsprotocol;
        }

        public static String[] parse(String s) {
            if (Arrays.stream(TlsProtocolVersion.values()).anyMatch(e -> e.protocol.equals(s))) {
                return s.split(",");
            }
            return null;
        }
    }
}
```
Allows an instance to be created and kept around and access to the parsed value.  This way, you can parse and validate exactly once:
```
  if (tlsEnabled && (tlsProtocolVersion != null)) {
            this.tlsProtocolVersion = new TLSProtocolVersion(tlsProtocolVersion);
            Preconditions.checkArgument(tlsProtocolVersion.tlsProtocols() != null,
                    "TLSProtocolVersion should be TLSv1.2 (strict 1.2), TLSv1.3 (strict 1.3), mixed mode");
  }
```

And at this point you have:
`this.tlsProtocolVersion.tlsProtocols()` --> returns `String[]` ready to go into the consumers.

Just taking GRPCServerConfigImpl as an example, the raw String from the config file comes in the constructor, but TLSProtocolVersion instance is what's kept around and referenced to by the lower level api calls that needs the String[]. 

GRPCServerConfigImpl.tlsProtocolVersion is of type `TLSProtocolVersion` essentially, not String.

Reply:
> Allows an instance to be created and kept around and access to the parsed value. This way, you can parse and validate exactly once:
I don;t see the need for a class here. I would recommend creating an array of enum instances from the String value passed to the Server early in the code (as early as possible) and storing the same in the Server Config and using this array of enums everywhere else, removing the need for further validation lower down.

Reply:
With the new changes, tlsprotocols are parsed early in the code (`Serviceconfig, Config`) classes and passed on to the respective components.